pagopa/io-app
# chore: [IOBP-316] Add return to origin page after payment flow completion

## Short description
This PR implements a new feature that allows users to automatically navigate back to their origin page once they have completed (successfully or not) the payment process. This enhancement improves user experience by providing a seamless transition back to their initial context after finishing a transaction.

## List of changes proposed in this pull request
- Added an additional argument into `walletPaymentInitState` action that expects an optional route name to go back to after the payment is completed/canceled.

## How to test
- Start a new payment flow until you reach the end of it;
- When you close the outcome message, you should be able to go back to the origin point that started the flow

## Preview

https://github.com/pagopa/io-app/assets/34343582/c4a42ab0-8676-4a7a-8631-624c3a98191a
